Add TensorModifyUpdate to document module (Java).

toregge opened the pull request:
@geirst: please review

Commit:
Add TensorModifyUpdate to document module (Java).

geirst commented on document/src/test/java/com/yahoo/document/update/SerializationTestCase.java:
```diff
@@ -57,4 +65,29 @@ public void testClearSerialization() {
+        update.addValueUpdate(new TensorModifyUpdate(TensorModifyUpdate.Operation.REPLACE, tensor));
+        GrowableByteBuffer buffer = serializeUpdate(update);
+        FieldUpdate deserializedUpdate = deserializeUpdate(buffer);
+        assertEquals("'tensorfield' [tensormodify replace TensorFieldValue {\n    classId: 1\n}\n]", deserializedUpdate.toString());
```
I would check the operation and tensor separately instead of depending on toString() format.

geirst commented on document/src/main/java/com/yahoo/document/update/TensorModifyUpdate.java:
```diff
@@ -0,0 +1,119 @@
+         * @param id the ID to search for
+         * @return the Operator with the specified ID, or null if it does not exist.
+         */
+        public static Operation getID(int id) {
```
Consider renaming to getOperation(). Also I think we can delete the function comment as it doesn't provide any extra value.

geirst commented on document/src/main/java/com/yahoo/document/update/TensorModifyUpdate.java:
```diff
@@ -0,0 +1,119 @@
+
+
+    public Operation getOperation() { return operation; }
+    public TensorFieldValue getTensor() { return tensor; }
```
Do we really need this when we have getValue() with the correct return type?

geirst reviewed (commented):
Overall looks good. Some comments to consider

Commit:
Adjust local variable names (operand -> tensor).
Remove unneded comments.
Don't use toString() to check for correct deserialization.

Commit:
Track abi change.

geirst reviewed (approved):
👍

geirst commented:
@lesters FYI
